### DISCUSSION TITLE
GitHub Enterprise - Monitoring Setup (AWS CloudWatch/Prometheus)

### DISCUSSION THREAD

**Discussion by kabilanpichaipillai** [community/community#39641] [Enterprise]
### Select Topic Area

Question

### Body

Currently GitHub Enterprise (GHE) is installed in AWS EC2 instance and looking for setup external monitoring such as CloudWatch Agent/Prometheus to track custom metrics - Disk used%, CPU Utilization, Memory used% etc.,
I was told that GHE license will be revoked if any 3rd party agent is installed in that GHE server. But I don't see any documents/reference about this. Please clarify below questionnaires,
 
1. Can AmazonCloudWatchAgent/Node Exporter be installed in GHE (linux) server to track custom metrics? Will there any problem to GHE license if any of these agent is installed in GHE server?
2. Is there anyway to track these metrics and move them to AWS Cloudwatch/Prometheus if those agents cannot be installed?

Note: During analysis, SNMP and Collectd can be used for external monitoring. But I am keenly looking for Cloudwatch/Prometheus integration with GHE server. Please do the needful!

**Comment by ossanna16** [comment]
Hi there @kabilanpichaipillai and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Reply by chetanyanamandra** [reply]
@kabilanpichaipillai did you figure this out?

**Reply by byrneh** [reply]
If I remember correctly there is a warning on administrative shell access login to not install additional software. 

Git support documentation wans the following 
> If you have installed additional third-party software on, or made modifications to, GitHub Enterprise Server outside of the recommendations of GitHub Enterprise Support, GitHub Enterprise Support may ask you to remove the software and/or modifications while attempting to resolve the issue. If the problem is fixed when the unsupported software or modifications have been removed

https://docs.github.com/en/enterprise-server@3.7/support/learning-about-github-support/about-github-support

You also need to be aware that GitHubs hotfix, patch and release upgrades are only designed and tested on the base images they provide.
